compiler/backend/con_to_decScript.sml

```diff
@@ -10,51 +10,58 @@ val _ = new_theory"con_to_dec"
  *)
 
 val _ = Define`
-  (init_globals [] idx =
-   Con NONE [])
+  (init_globals tra tidx [] idx =
+   Con (mk_cons tra tidx) NONE [])
   ∧
-  (init_globals (x::vars) idx =
-   Let NONE (App (Init_global_var idx) [Var_local x]) (init_globals vars (idx+1)))`;
+  (init_globals tra tidx (x::vars) idx =
+   Let (mk_cons tra tidx) NONE (App (mk_cons tra (tidx+1)) (Init_global_var idx)
+   [Var_local (mk_cons tra (tidx+2)) x]) (init_globals tra (tidx+3) vars (idx+1)))`;
 
 
 val _ = Define `
-  (init_global_funs next [] = Con NONE [])
+  (init_global_funs tra tidx next [] = Con (mk_cons tra tidx) NONE [])
   ∧
-  (init_global_funs next ((f,x,e)::funs) =
-   Let NONE (App (Init_global_var next) [Fun x e]) (init_global_funs (next+1) funs))`;
-
+  (init_global_funs tra tidx next ((f,x,e)::funs) =
+   Let (mk_cons tra tidx) NONE (App (mk_cons tra (tidx+1)) (Init_global_var next) [Fun (mk_cons tra (tidx+2)) x e]) (init_global_funs tra (tidx+3) (next+1) funs))`;
+(*TODO: We should check whether introducing a trace to Dletrec and is sensible, in
```

Comment:
Change this TODO to be about us figuring out a "good enough" trace for the declarations. The mailing list decided we should not add traces to declarations, since this means updating the proofs again.